File: python/hamming/hamming.py
"""
hamming - Calculate the Hamming difference between two DNA strands
"""

def distance(dna_1, dna_2):
    if len(dna_1) != len(dna_2):
        raise ValueError('DNA strands not same length!')
    else:
        return sum([d1 != d2 for d1, d2 in zip(dna_1, dna_2)])

File: javascript/hamming/hamming.js
/*
hamming - Calculate the Hamming difference between two DNA strands.
*/

var Hamming = function() {};

Hamming.prototype.compute = function(dna1, dna2) {
  if (dna1.length !== dna2.length) {
    throw new Error('DNA strands must be of equal length.');
  }

  var distance = 0;

  // treat string as array and loop through comparing each value
  for (var i = 0; i < dna1.length; i++) {
      if (dna1.charAt(i) !== dna2.charAt(i)) {
        distance++;
      }
  }

  return distance;
}

module.exports = Hamming;
